frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx

```diff
@@ -0,0 +1,239 @@
+            {!!errorInfo.localizedEntityName && (
+              <FormCheckbox
+                name="entityInfo"
+                label={t`${capitalize(
+                  errorInfo.localizedEntityName,
+                )} definition`}
+              />
+            )}
+            <FormCheckbox
+              name="frontendErrors"
+              label={t`Browser error messages`}
+            />
+            {!!errorInfo?.logs && (
+              <>
+                <FormCheckbox
+                  name="backendErrors"
+                  label={t`All server error messages`}
+                />
+                <FormCheckbox name="logs" label={t`All server logs`} />
+                <FormCheckbox
+                  name="userLogs"
+                  label={t`Server logs from the current user only`}
+                />
+              </>
+            )}
+            <FormCheckbox
+              name="instanceInfo"
+              // eslint-disable-next-line no-literal-metabase-strings -- we're mucking around in the software here
+              label={t`Metabase instance version and settings`}
+            />
+          </Stack>
+          <Alert variant="warning">
+            {t`Review the downloaded file before sharing it, as the diagnostic info may contain sensitive data.`}
+          </Alert>
+          <Flex gap="sm" justify="flex-end" mt="lg">
+            <Button onClick={onClose}>{t`Cancel`}</Button>
+            <FormSubmitButton
+              variant="filled"
+              leftIcon={<Icon name="download" />}
+              label={t`Download`}
+              color="brand"
+            />
+          </Flex>
+        </Form>
+      </FormProvider>
+    </Modal>
+  );
+};
+
+export const ErrorDiagnosticModalTrigger = () => {
+  const { value: errorInfo, loading, error } = useErrorInfo();
+
+  const [isModalOpen, setModalOpen] = useState(false);
+
+  if (loading || !errorInfo) {
+    return (
+      <Center>
+        <Loader />
+      </Center>
+    );
+  }
+
+  if (error) {
+    console.error(error);
+    return null;
+  }
+
+  return (
+    <ErrorBoundary>
+      <Stack justify="center" my="lg">
+        <Box>
+          <Text align="center">
+            {jt`Click the button below to download diagnostic information to send
+            to
+            ${(
+              <Link key="email" variant="brand" to="mailto:help@metabase.com">
+                {t`help@metabase.com`}
+              </Link>
+            )}`}
+          </Text>
+        </Box>
+        <Button
+          leftIcon={<Icon name="download" />}
+          onClick={() => setModalOpen(true)}
+        >
+          {t`Download diagnostic information`}
+        </Button>
+      </Stack>
+      {isModalOpen && (
+        <ErrorDiagnosticModal
+          errorInfo={errorInfo}
+          onClose={() => setModalOpen(false)}
+        />
+      )}
+    </ErrorBoundary>
+  );
+};
+
+export function KeyboardTriggeredErrorModal() {
+  const [
+    isShowingDiagnosticModal,
```

Comment:
"shouldShowDiagnosticModal" might be more readable because it makes clear that this variable determines the modal's state. It does not record whether the modal is already open

Reply:
I think this is contrary to existing patterns, and "should" often implies a function, rather than a value